### ISSUE TITLE
Plugins: make it possible to make an app plugin not accessible directly (by URL)

### ISSUE THREAD

**Issue by leventebalogh** [grafana/grafana#49937]
### What's the problem?
In case an app plugin is moved to another section entirely (or all of it's sub-pages are moved to different sections) it doesn't make sense anymore to display it under `/a/:plugin-id` and in the navigation tree, as this can cause confusion.
Based on the comments we would also like to have a redirect from the default `/a/:plugin-id` URL, so using the old link doesn't result in a 404.

**Default plugin route**
To be honest probably we should leave in the default route for `/a/:plugin-id` at least until Grafana 10, and make it behave like this:
- **A)** no app-plugin installed and enabled with the same ID -> 404 page
- **B)** app-plugin found & has default page -> render default page
- **C)** app-plugin found & no default page
    - **C.1)** whole app-plugin moved under a different section -> redirect to the app-plugin's new path
   -   **C.2)** app-plugin's standalone pages moved separately -> redirect to the first page's new path 

---

### Solution 1 ._(takes longer but cleaner)_

1. Update the configuration options for plugin-navigation as outlined in [this design doc](https://docs.google.com/document/d/1eDZcmRUdb5s8S1rdFejRJGLZ98VE_RVQtpQnNPa6Eag/edit#)
2. Update the apps that already use this configuration or that want to be rendered under a certain path even though they don't have any pages registered (e.g. "cloud-home-app")
3. Implement the app plugin default route redirect as outlined under "Default plugin route" above 

### Solution 2. _(probably quicker but dirtier)_

1. Add a page to the `"cloud-home-app"` that is not visible in the navigation, but allows to render it under a certain path
2. Implement the app plugin default route redirect as outlined under "Default plugin route" above 
3. Update the navigation config as outlined [in this design doc](https://docs.google.com/document/d/1eDZcmRUdb5s8S1rdFejRJGLZ98VE_RVQtpQnNPa6Eag/edit#) sometime in the future

**Comment by jarben** [comment]
cc @rrmf this will affect the sign up app & the onboarding flow URL links

We have discussed this today with @mikkancso and @anaivanov and it seems like the redirect would be a preferred option in case some users still use old links or bookmarks.

**Comment by leventebalogh** [comment]
Thanks @jarben, that's a good point, we should handle old URLs with redirects.

I will update the description of the issue today to be more clear on how I think we could approach this in a way that is feasible and backwards compatible.

**Comment by mikkancso** [comment]
We talked about this a few days ago with @leventebalogh , and we agreed that we don't really have to change the navigation logic in Grafana. Instead, we should do this:

1. There is a new top section called "Connections" in Grafana coming with the topnav feature, which would just show the onboarding plugin. We should definitely not show this top section when our feature toggle `dataConnectionsConsole` is turned on.
2. The onboarding plugin should redirect from `/a/grafana-easystart-plugin` to `/connections/connect-data` when the feature is on.

This way we can keep our changes simple and still reach the goal of having exactly one place in the navigation where the plugin renders its pages.

**Comment by mikkancso** [comment]
> 1. There is a new top section called "Connections" in Grafana coming with the topnav feature, which would just show the onboarding plugin. We should definitely not show this top section when our feature toggle dataConnectionsConsole is turned on.

I originally meant that we can just put [this logic](https://github.com/grafana/grafana/blob/35090c376ca3643d930123455e2cb5eaa8bdf1e4/pkg/services/navtree/navtreeimpl/applinks.go#L265) behind our feature toggle, but it's not that simple. If this piece of logic is not present, then the plugin will be put under the "Apps" section in the navtree.

So we need another way to hide the "main page" of our plugin from the navtree.

**Comment by mikkancso** [comment]
What are our goals?

**Goal 1**: work in current situation, TopNav off, DataConnectionsConsole off
- plugin should have its default nav entry under "Integrations and connections"
- plugin should not have other pages in the navtree

**Goal 2**: work with TopNav on, DataConnectionsConsole off
- plugin should have its default nav entry moved to "Connections" top section
- plugin should not have other pages in the navtree

**Goal 3**: work with TopNav on, DataConnectionsConsole on
- plugin should not have its default nav entry anywhere (not in the previous "Connections", neither under "Apps")
- plugin should have other pages under our "Connections" section
- plugin should redirect from `/a/grafana-easystart-app` to `/connections/connect-data`

The current code satisfies Goals 1&2. Eventually, after we rolled out Connections to everyone, I believe we will be able to remove the deafult page entry from the [plugin.json](https://github.com/grafana/cloud-onboarding/blob/24a5e4aa7da5a5602cf16b68052eb9a4fcb4ee31/plugin/src/plugin.json#L160-L167).

But, we need a mechanism to hide that page depending on our feature toggle, DataConnectionsConsole.
After looking at [this piece of code](https://github.com/grafana/grafana/blob/35090c376ca3643d930123455e2cb5eaa8bdf1e4/pkg/services/navtree/navtreeimpl/applinks.go#L107-L135), I realized that we can make the default page to be left out of the navtree if we add this line to Grafana's config:
```
"/a/grafana-easystart-app" = nonexistent
```

This is quite a hacky solution, but we could make this configuration explicitly supported (even later). But I quite like it, because we can roll this out by just relying on having this line in the config at the same time when the feature toggle is turned on, and no code change is strictly necessary.
